core/src/main/java/io/questdb/griffin/SqlKeywords.java

```diff
@@ -504,24 +645,38 @@ public static boolean isHourKeyword(CharSequence tok) {
                 && (tok.charAt(i)) == 'r';
     }
 
-    public static boolean isInKeyword(CharSequence tok) {
+    public static boolean isIfKeyword(CharSequence tok) {
         if (tok.length() != 2) {
             return false;
         }
 
         int i = 0;
         return (tok.charAt(i++) | 32) == 'i'
+                && (tok.charAt(i) | 32) == 'f';
+    }
+
+    public static boolean isWithinKeyword(CharSequence tok) {
+        if (tok == null || tok.length() != 6) {
+            return false;
+        }
+
+        int i = 0;
+        return (tok.charAt(i++) | 32) == 'w'
+                && (tok.charAt(i++) | 32) == 'i'
+                && (tok.charAt(i++) | 32) == 't'
+                && (tok.charAt(i++) | 32) == 'h'
+                && (tok.charAt(i++) | 32) == 'i'
                 && (tok.charAt(i) | 32) == 'n';
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 664:669, 361:366.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.